src/block_cache/block_chunkcache.c

```diff
@@ -510,15 +554,26 @@ __wt_chunkcache_setup(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig
         chunkcache->type = WT_CHUNKCACHE_IN_VOLATILE_MEMORY;
     else if (WT_STRING_MATCH("file", cval.str, cval.len) ||
       WT_STRING_MATCH("FILE", cval.str, cval.len)) {
-#ifdef ENABLE_MEMKIND
+
         chunkcache->type = WT_CHUNKCACHE_FILE;
-        WT_RET(__wt_config_gets(session, cfg, "chunk_cache.device_path", &cval));
-        WT_RET(__wt_strndup(session, cval.str, cval.len, &chunkcache->dev_path));
-        if (!__wt_absolute_path(chunkcache->dev_path))
-            WT_RET_MSG(session, EINVAL, "File directory must be an absolute path");
-#else
-        WT_RET_MSG(session, EINVAL, "chunk cache of type FILE requires libmemkind");
-#endif
+        WT_RET(__wt_config_gets(session, cfg, "chunk_cache.storage_path", &cval));
+        WT_RET(__wt_strndup(session, cval.str, cval.len, &chunkcache->storage_path));
+        if (!__wt_absolute_path(chunkcache->storage_path))
+            WT_RET_MSG(session, EINVAL, "Storage location must be an absolute path");
+
+        WT_RET(__wt_open(session, chunkcache->storage_path, WT_FS_OPEN_FILE_TYPE_DATA,
+          WT_FS_OPEN_CREATE | WT_FS_OPEN_FORCE_MMAP, &chunkcache->fh));
+
+        WT_RET(chunkcache->fh->handle->fh_truncate(
+          chunkcache->fh->handle, &session->iface, (wt_off_t)chunkcache->capacity));
+
+        WT_RET(chunkcache->fh->handle->fh_map(chunkcache->fh->handle, &session->iface,
+          (void **)&chunkcache->memory, &mapped_size, NULL));
+        WT_ASSERT_ALWAYS(session, mapped_size == chunkcache->capacity,
```

Comment:
I'm rethinking whether this should be an assertion. If the user asks for an insanely large chunk cache, or we have a bad fragmentation problem, I don't think that justifies killing the process. Perhaps an ENOMEM?

Reply:
I understand, done so here: [c7ee45f](https://github.com/wiredtiger/wiredtiger/pull/9442/commits/c7ee45f7923d2d375c66f92a44eff9157fdf8b7f)